stl/inc/vector

```diff
@@ -458,6 +494,139 @@ public:
     using difference_type = typename _Alty_traits::difference_type;
 
 private:
+#ifdef _INSERT_VECTOR_ANNOTATION
+    static _CONSTEXPR20 void _Create_annotation(const vector& _Myvec_) noexcept {
+        // Annotates the shadow memory of the valid range
+        auto& _My_data = _Myvec_._Mypair._Myval2;
+        _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Myend, _My_data._Mylast);
+    }
+
+    static _CONSTEXPR20 void _Clear_annotation(const vector& _Myvec_) noexcept {
+        // Removes the shadow memory annotation of the range

```

Comment:
Naming nitpick: I think I understand what this is doing (telling ASAN that the entire range of memory is valid, because as @CaseyCarter noted, we don't know what the allocator is going to do during deallocation). However, the appearance of "clear" in the name led me to expect `vector::clear`-style behavior (which destroys all elements, i.e. nobody is valid after that). After thinking about it a little, I am increasingly concerned about the potential for confusion given Create/Clear/Change all starting with the same letter. Could these be renamed to look more distinctive, and avoid being too similar to `vector` member functions? For example, Create/Remove/Modify would work, I think. Also affects the macros.